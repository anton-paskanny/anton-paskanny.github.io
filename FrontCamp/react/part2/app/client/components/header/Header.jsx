import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';

import './styles.css';


export default class Header extends React.Component {
  constructor(props) {
    super(props);

    this.logOutHandler = this.logOutHandler.bind(this);
  }
  logOutHandler(event) {
    event.preventDefault();

    this.props.logOut();
  }
  render() {
    return (
      <header className="header">
        <Link className="header__title" to="/">
          <h2>React app</h2>
        </Link>
        <nav className="header__nav">
          {
            !this.props.isLoggedIn &&
            <Link className="header__link hoverable" to="/signin">Sign In</Link>
          }
          {
            !this.props.isLoggedIn &&
            <Link className="header__link hoverable" to="/signup">Sign Up</Link>
          }
          {
            this.props.isLoggedIn &&
            <a className="header__link hoverable" href="#" onClick={this.logOutHandler}>Log Out</a>
          }
        </nav>
      </header>
    )
  }
}
