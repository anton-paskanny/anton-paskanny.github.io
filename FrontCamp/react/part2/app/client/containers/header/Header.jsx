import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { setLoggedIn } from '../../actions/user';
import { logoutUser } from '../../services/user';

import './styles.css';


class Header extends React.Component {
  constructor(props) {
    super(props);

    this.logOutHandler = this.logOutHandler.bind(this);
  }
  logOutHandler(event) {
    event.preventDefault();

    logoutUser().then(res => {
      this.props.setLoggedIn(res.user);
    });
    
    // fetch('http://localhost:3000/users/logout', {
    //   credentials: 'include'
    // })
    // .then(res => res.json())
    // .then(res => {
    //   console.log('Logout handler: ', res);
    //   this.props.setLoggedIn(res.user);
    // });
  }
  render() {
    return (
      <header className="header">
        <Link className="header__title" to="/">
          <h2>React app</h2>
        </Link>
        <nav className="header__nav">
          {
            !this.props.isLoggedIn &&
            <Link className="header__link" to="/signin">Sign In</Link>
          }
          <Link className="header__link" to="/signup">Sign Up</Link>
          {
            this.props.isLoggedIn &&
            <Link className="header__link" to="/blogs">Blogs</Link>
          }
          {
            this.props.isLoggedIn &&
            <a className="header__link" href="#" onClick={this.logOutHandler}>Log Out</a>
          }
        </nav>
      </header>
    )
  }
}

const mapStateToProps = state => ({
  isLoggedIn: state.user.isLoggedIn,
  user: state.user.data
});

const mapDispatchToProps = dispatch => ({
  setLoggedIn(user) {
    dispatch(setLoggedIn(user));
  }
});

export default connect(mapStateToProps, mapDispatchToProps)(Header);
