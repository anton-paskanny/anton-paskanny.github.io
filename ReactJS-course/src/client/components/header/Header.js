import React from 'react';
import { withRouter } from 'react-router-dom'

import SearchForm from '../../containers/SearchForm';

import styles from './styles.css';

const SearchFormWithRouter = withRouter(({ history }) => <SearchForm history={history} />);

const Header = () => (
    <header className="main-header">
        <div className="main-header__inner-wrapper">
            <h1 className="main-header__title common-title">netflixroulette</h1>
            <SearchFormWithRouter />
        </div>
    </header>
);

export default Header;