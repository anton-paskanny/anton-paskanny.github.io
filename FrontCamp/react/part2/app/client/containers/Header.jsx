import { connect } from 'react-redux';
import { logOut } from '../actions/user';

import Header from '../components/header/Header.jsx';


const mapStateToProps = state => ({
  isLoggedIn: state.user.isLoggedIn,
  user: state.user.data
});

const mapDispatchToProps = dispatch => ({
  logOut() {
    dispatch(logOut());
  }
});

export default connect(mapStateToProps, mapDispatchToProps)(Header);
