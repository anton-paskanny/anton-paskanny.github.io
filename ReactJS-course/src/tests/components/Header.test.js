import React from 'react';
import { shallow } from 'enzyme';

import Header from '../../client/components/Header/Header';

describe('Header', () => {
    it('render component correctly', () => {
        const component = shallow(<Header />);
        expect(component).toMatchSnapshot();
    });
});



