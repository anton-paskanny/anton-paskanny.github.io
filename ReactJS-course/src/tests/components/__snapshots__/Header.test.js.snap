// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header render component correctly 1`] = `
<header
  className="main-header"
>
  <div
    className="main-header__inner-wrapper"
  >
    <h1
      className="main-header__title common-title"
    >
      netflixroulette
    </h1>
    <Connect(Search) />
  </div>
</header>
`;
